WordPress/gutenberg
# Patterns: show theme patterns from directory in site editor

## What?
Shows theme patterns specified in theme.json in the site editor.

## Why?
Currently patterns specified in the `patterns` section of `theme.json` display in the post editor site editor, but do not disply in the site editor patterns page.

Fixes: #55819

## How?
Removes `pattern-directory/theme`  from the `PATTERN_CORE_SOURCES` constant that is used to filter these patterns in the site editor.

## Testing Instructions

- Activate TT4 theme in a site
- In the Post Editor open the insterter patterns tab and the `Testimonial` tab and make sure two patterns appear - `Centered Terstimonial` and `Clients`
- In the site editor patterns page open the `Testimonial` category and make sure the same two patterns appear
- In the site editor try duplicating the `Clients` pattern and make sure this works as expected and new pattern is usable/editable

## Screenshots or screencast 
Before:

https://github.com/WordPress/gutenberg/assets/3629020/fa14cfe8-c4bd-4920-8c04-bd2b69af9892

After:

https://github.com/WordPress/gutenberg/assets/3629020/174da87a-d4ed-435e-aa51-8823aa62272f
